paimon-core/src/main/java/org/apache/paimon/mergetree/compact/PartialUpdateMergeFunction.java

```diff
@@ -106,36 +116,23 @@ public void add(KeyValue kv) {
 
         latestSequenceNumber = kv.sequenceNumber();
         isEmpty = false;
-        if (fieldSequences.isEmpty()) {
-            updateNonNullFields(kv);
-        } else {
-            updateWithSequenceGroup(kv);
-        }
+        update(kv);
     }
 
-    private void updateNonNullFields(KeyValue kv) {
-        for (int i = 0; i < getters.length; i++) {
-            Object field = getters[i].getFieldOrNull(kv.value());
-            if (field != null) {
-                row.setField(i, field);
-            }
-        }
-    }
-
-    private void updateWithSequenceGroup(KeyValue kv) {
+    private void update(KeyValue kv) {
         for (int i = 0; i < getters.length; i++) {
             Object field = getters[i].getFieldOrNull(kv.value());
             SequenceGenerator sequenceGen = fieldSequences.get(i);
+            FieldAggregator aggregator = fieldAggregators.get(i);
+            Object accumulator = getters[i].getFieldOrNull(row);
             if (sequenceGen == null) {
-                if (field != null) {
-                    row.setField(i, field);
-                }
+                row.setField(i, aggregator.agg(accumulator, field));
             } else {
                 Long currentSeq = sequenceGen.generateNullable(kv.value());
                 if (currentSeq != null) {
                     Long previousSeq = sequenceGen.generateNullable(row);
                     if (previousSeq == null || currentSeq >= previousSeq) {
-                        row.setField(i, field);
+                        row.setField(i, aggregator.agg(accumulator, field));
```

Comment:
This is strange, we may introduce `FieldAggregator.aggForOldSequence(...)`, for example, sum should be also agg if it is a late record?

There should be tests..

Reply:
You are right, will fix this

Reply:
how about first_value / first_not_null_value?

Reply:
This will be done in the `first_value` agg function. For the `first_value`,  when an older sequence record in , it should updated to the new value.